centreon/src/Core/ResourceAccess/Application/UseCase/AddRule/AddRule.php

```diff
@@ -106,11 +106,28 @@ public function __invoke(
                 // At least one ID must be provided for contact or contactgroup
                 $this->validator->assertContactsAndContactGroupsAreNotEmpty(
                     $request->contactIds,
-                    $request->contactGroupIds
+                    $request->contactGroupIds,
+                    $request->applyToAllContacts,
+                    $request->applyToAllContactGroups
                 );
 
-                $this->validator->assertContactIdsAreValid($request->contactIds);
-                $this->validator->assertContactGroupIdsAreValid($request->contactGroupIds);
+                /**
+                 * Contact and ContactGroup IDs need validation only if IDs are provided and that all property is not
+                 * set to true.
+                 */
+                if (
+                    ! $request->applyToAllContacts
+                    && $request->contactIds !== []
+                ) {
```

Comment:
Just Suggest:

```suggestion
                if (! $request->applyToAllContacts) {
```

if the array is empty it will not break your validator `array_diff($contactIds, $validIds)`
$contactIds = [], $validIds = []